apps/service_providers/llm_service/state.py

```diff
@@ -184,9 +184,25 @@ def get_assistant_instructions(self):
         # Langchain doesn't support the `additional_instructions` parameter that the API specifies, so we have to
         # override the instructions if we want to pass in dynamic data.
         # https://github.com/langchain-ai/langchain/blob/cccc8fbe2fe59bde0846875f67aa046aeb1105a3/libs/langchain/langchain/agents/openai_assistant/base.py#L491
-        return self.experiment.assistant.instructions.format(
-            participant_data=self.get_participant_data(), current_datetime=self.get_current_datetime()
+        instruction_prompt = self.experiment.assistant.instructions
+        if self.experiment.assistant.include_file_info:
+            instruction_prompt += "\n\nFile type information:\n{file_type_info}"
+
+        instructions = instruction_prompt.format(
+            participant_data=self.get_participant_data(),
+            current_datetime=self.get_current_datetime(),
+            file_type_info=self.get_file_type_info(),
         )
+        return instructions
+
+    def get_file_type_info(self):
+        if not self.experiment.assistant.include_file_info:
+            return ""
+        attachments = self.session.chat.attachments.filter(tool_type__in=["code_interpreter"])
+        file_type_info = []
+        for att in attachments:
+            file_type_info.extend([{file.external_id: file.content_type} for file in att.files.all()])
```

Comment:
Is the assistant able to match the file ID to the file?

Reply:
From what I could tell, yes. Example:

### In this chat we append file types
![image](https://github.com/user-attachments/assets/f6efb6e6-ef07-499a-be90-aa53180ccda6)

### In this chat we don't append file types
Notice where it says "The presence of the word directory suggests that this file might be a Word document (.docx)". It only figures out it's a word doc later on

![image](https://github.com/user-attachments/assets/425f8dfe-e9a4-4627-9db9-024344ada03c)